languages/rust/oso/src/host/class.rs

```diff
@@ -94,14 +115,37 @@ impl<T> Class<T> {
 
     pub fn set_constructor<F, Args>(mut self, f: F) -> Self
     where
-        T: 'static,
         F: Function<Args, Result = T> + 'static,
         Args: FromPolar + 'static,
     {
         self.constructor = Some(Constructor::new(f));
         self
     }
 
+    pub fn set_equality_check<F>(mut self, f: F) -> Self
+    where
+        F: Fn(&T, &T) -> bool + 'static,
+        T: std::fmt::Debug
+    {
+        self.equality_check = Arc::new(move |a, b| {
+            println!("equality check");
+
+            let a = downcast(a)?;
+            let b = downcast(b)?;
+
+            println!("{:?} == {:?}", a, b);
+
+            Ok((f)(a, b))
+        });
+
+        self
+    }
+
+    pub fn with_equality_check(self) -> Self
+    where T: PartialEq<T> + std::fmt::Debug {
```

Comment:
This is really cool, but why is debug needed?

Reply:
Removed, was just temporary.  Separate follow up for actually being able to debug print.